tools/tinytest-codegen.py

```diff
@@ -50,58 +79,47 @@ def script_to_map(test_file):
 testgroup_member = '  {{ "{name}", {name}_tests }},'
 
 ## XXX: may be we could have `--without <groups>` argument...
-# currently these tests are selected because they pass on qemu-arm
-test_dirs = (
-    "basics",
-    "micropython",
-    "misc",
-    "extmod",
-    "float",
-    "inlineasm",
-    "qemu-arm",
-)  # 'import', 'io',)
-exclude_tests = (
-    # pattern matching in .exp
-    "basics/bytes_compare3.py",
-    "extmod/ticks_diff.py",
-    "extmod/time_ms_us.py",
-    # unicode char issue
-    "extmod/json_loads.py",
-    # doesn't output to python stdout
-    "extmod/re_debug.py",
-    "extmod/vfs_basic.py",
-    "extmod/vfs_fat_ramdisk.py",
-    "extmod/vfs_fat_fileio.py",
-    "extmod/vfs_fat_fsusermount.py",
-    "extmod/vfs_fat_oldproto.py",
-    # rounding issues
-    "float/float_divmod.py",
-    # requires double precision floating point to work
-    "float/float2int_doubleprec_intbig.py",
-    "float/float_format_ints_doubleprec.py",
-    "float/float_parse_doubleprec.py",
-    # inline asm FP tests (require Cortex-M4)
-    "inlineasm/asmfpaddsub.py",
-    "inlineasm/asmfpcmp.py",
-    "inlineasm/asmfpldrstr.py",
-    "inlineasm/asmfpmuldiv.py",
-    "inlineasm/asmfpsqrt.py",
-    # different filename in output
-    "micropython/emg_exc.py",
-    "micropython/heapalloc_traceback.py",
-    # don't have emergency exception buffer
-    "micropython/heapalloc_exc_compressed_emg_exc.py",
-    # pattern matching in .exp
-    "micropython/meminfo.py",
-    # needs sys stdfiles
-    "misc/print_exception.py",
-    # settrace .exp files are too large
-    "misc/sys_settrace_loop.py",
-    "misc/sys_settrace_generator.py",
-    "misc/sys_settrace_features.py",
-    # don't have f-string
-    "basics/string_fstring.py",
-    "basics/string_fstring_debug.py",
+
+test_dirs = set(
+    (
+        "basics",
```

Comment:
I'm a little bit unsure how this list got divided. Why isn't `micropython` in the base set?

Reply:
And therefore, the `micropython/*` exclusions could move back.

Reply:
That's a mistake I made when changing from pre-baked per port lists to a base plus extras model. Thanks for pointing that out.